src/main/java/uk/gov/companieshouse/registeredemailaddressapi/service/MockEmailRetrieverImpl.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.companieshouse.registeredemailaddressapi.service;
+
+import static java.lang.String.format;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.company.RegisteredEmailAddressJson;
+import uk.gov.companieshouse.registeredemailaddressapi.exception.ServiceException;
+import uk.gov.companieshouse.registeredemailaddressapi.utils.ApiLogger;
+
+import java.util.HashMap;
+
+@Service
+@Qualifier("mockEmailRetrieverImpl")
+@ConditionalOnExpression("'${env.name}'.equals('stagsbox') || '${env.name}'.equals('livesbox')")
```

Comment:
I understand how this works in conjunction with the conditional in the Oracle impl, pity it is not possible to have one as "default" (the Oracle one) and other (this mock) as an override